jc/parsers/iftop.py

```diff
@@ -0,0 +1,397 @@
+                    },
+                    {
+                        "host_name": "10.10.15.72",
+                        "host_port": "40876",
+                        "last_2s": "208b",
+                        "last_10s": "208b",
+                        "last_40s": "208b",
+                        "cumulative": "52B",
+                        "direction": "receive"
+                    }
+                ]
+            }
+        ],
+        "total_send_rate": {
+            "last_2s": "448b",
+            "last_10s": "448b",
+            "last_40s": "448b"
+        },
+        "total_receive_rate": {
+            "last_2s": "208b",
+            "last_10s": "208b",
+            "last_40s": "208b"
+        },
+        "total_send_and_receive_rate": {
+            "last_2s": "656b",
+            "last_10s": "656b",
+            "last_40s": "656b"
+        },
+        "peak_rate": {
+            "last_2s": "448b",
+            "last_10s": "208b",
+            "last_40s": "656b"
+        },
+        "cumulative_rate": {
+            "last_2s": "112B",
+            "last_10s": "52B",
+            "last_40s": "164B"
+        }
+    }
+    ]
+      
+"""
+import re
+from typing import List, Dict
+from jc.jc_types import JSONDictType
+import jc.utils
+
+
+class info:
+    """Provides parser metadata (version, author, etc.)"""
+
+    version = "0.1"
+    description = "`iftop` command parser"
+    author = "Ron Green"
+    author_email = "11993626+georgettica@users.noreply.github.com"
+    compatible = ["linux"]
+    tags = ["command"]
+
+
+__version__ = info.version
+
+
+# def _process(proc_data: List[JSONDictType], quiet: bool = False) -> List[JSONDictType]:
+#     """
+#     Final processing to conform to the schema.
+
+#     Parameters:
+
+#         proc_data:   (List of Dictionaries) raw structured data to process
+
+#     Returns:
+
+#         List of Dictionaries. Structured to conform to the schema.
+#     """
+#     return proc_data
+
+
+def parse(data: str, raw: bool = False, quiet: bool = False) -> List[JSONDictType]:
+    """
+    Main text parsing function
+
+    Parameters:
+
+        data:        (string)  text data to parse
+        raw:         (boolean) unprocessed output if True
+        quiet:       (boolean) suppress warning messages if True
+
+    Returns:
+
+        List of Dictionaries. Raw or processed structured data.
+    """
+    jc.utils.compatibility(__name__, info.compatible, quiet)
+    jc.utils.input_type_check(data)
+
+    raw_output: List[Dict] = []
+
+    # for backwards compatibility, preset all fields to None
+    interface_obj: Dict = {
+        "device": None,
+        "ip_address": None,
+        "mac_address": None,
+        "clients": None,
+        "total_send_rate": None,
+        "total_receive_rate": None,
+        "total_send_and_receive_rate": None,
+        "peak_rate": None,
+        "cumulative_rate": None,
+    }
+
+    interface_item: Dict = interface_obj.copy()
```

Comment:
This can be a normal empty dict. I was just copying in the other parser so it would keep the default values for backward compatibility.